tests/foreman/cli/test_host_collection.py

```diff
@@ -28,258 +28,247 @@
 from robottelo.constants import DEFAULT_CV
 from robottelo.constants import ENVIRONMENT
 from robottelo.datafactory import invalid_values_list
+from robottelo.datafactory import parametrized
 from robottelo.datafactory import valid_data_list
-from robottelo.test import CLITestCase
 
 
-class HostCollectionTestCase(CLITestCase):
-    """Tests for Host Collections via Hammer CLI"""
-
-    @classmethod
-    def setUpClass(cls):
-        """Prepare some data to be used in tests"""
-        super().setUpClass()
-        cls.organization = make_org()
-        cls.library = LifecycleEnvironment.info(
-            {'organization-id': cls.organization['id'], 'name': ENVIRONMENT}
+@pytest.fixture(scope='module', autouse=True)
+def set_up_a_host_collection(module_org):
+    """ Make a host collection for use in tests"""
+    make_host_collection({'organization-id': module_org.id})
+
+
+def _make_fake_host_helper(module_org):
```

Comment:
Since this is only dependent on the module_org, I believe it can be defined as a function scoped fixture. 

You can indirectly parametrize the 'count' of hosts, for the test case where you need two instead of one. The fixture can yield a list of host(s) for consistency. It could also delete the fake host after yielding, to clean up after itself.

This is an **optional** change - its borderline whether you'd call this 'setup' for the test. Its also questionable whether its worth the indirect parametrization instead of just calling the function in a loop when you need multiple hosts.

Reply:
@mshriver 

> Since this is only dependent on the module_org, I believe it can be defined as a function scoped fixture.
> 
That works OK for the end to end test.

I cannot call a fixture twice from one test and could not find an easy way around this.

> You can indirectly parametrize the 'count' of hosts, for the test case where you need two instead of one. 

By "indirectly parametrize" do you mean something like:
@pytest.mark.parametrize('fake_host', [2], indirect=['fake_host'])

The fixture can yield a list of host(s) for consistency. It could also delete the fake host after yielding, to clean up after itself.
> 
> This is an **optional** change - its borderline whether you'd call this 'setup' for the test. Its also questionable whether its worth the indirect parametrization instead of just calling the function in a loop when you need multiple hosts.

Only two tests in this module use the ` _make_fake_host_helper`,  so its not really "setup for tests"; we do not need it to be run in advance.

Not sure if this is worth it.

Thank you